File: 0841-shortest-distance-to-a-character/0841-shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> list[int]:
        n = len(s)
        result = [0] * n
        prev = float('-inf')  # Use negative infinity for the left-to-right pass

        # Left to right pass
        for i in range(n):
            if s[i] == c:
                prev = i
            result[i] = i - prev

        prev = float('inf')  # Use positive infinity for the right-to-left pass

        # Right to left pass
        for i in range(n - 1, -1, -1):
            if s[i] == c:
                prev = i
            result[i] = min(result[i], prev - i)

        return result

File: 0841-shortest-distance-to-a-character/0841-shortest-distance-to-a-character.java
class Solution {
    public int[] shortestToChar(String s, char c) {
        int n=s.length();
        int [] arr=new int[n];
        for(int i=0;i<n;i++){
            int mindist=Integer.MAX_VALUE;
            for(int j=0;j<n;j++){
                if(s.charAt(j)==c){
                    mindist=Math.min(mindist,Math.abs(i-j));

                }
            }
            arr[i]=mindist;
        }
        return arr;
    }
}